### PR TITLE
refactor: remove mui overrides in mui-setup

### PR DESCRIPTION
**Pull request by haxxmaxx**
Beginning to refactor to remove the overrides in `mui-setup`. We are applying styles in so many places. It's decided to use styled whenever possible. It creates the cleanest component files and is faster than sx. The overrides in mui-setup where really hard to read also. Keeping `MuiTable.styleOverrides.root = {}` since it contains a lot of stuff we don't want

This refactor slightly changes the look of the selects. But those will look different when we actually change the styling, so I am updating the artifacts for that

### CODE REVIEW

File: src/table/styles.ts [hunk 1]
```diff
@@ -102,9 +136,21 @@ export const StyledHeadRow = styled(TableRow, {
   },
 }));
 
+export const StyledHeadCell = styled(TableCell)({
+  ...cellCommon,
+  lineHeight: '150%',
+});
```
Comment (Donya-qlik) [inline src/table/styles.ts:142]: I've added the similar style in my PR but never mind I'll update it with yours after you merge it.

**Review by Donya-qlik — APPROVED** [review]
Totally agree with the refactor you made over styles 👍 
Just a commented code line you missed to remove. 
LGTM!

File: src/table/styles.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/table/styles.ts:142]: ok

File: src/table/styles.ts [hunk 2]
```diff
@@ -45,6 +59,7 @@ export const StyledSelect = styled(Select, {
   shouldForwardProp: (prop: string) => prop !== 'paginationTheme',
 })(({ paginationTheme }) => ({
   backgroundColor: 'inherit',
+  // padding: '0px 12px',
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/styles.ts:62]: I have another PR ready already. i'll do it in the next one

**Pull request merged**